docs/howto/charm-to-poetry.rst

```diff
@@ -0,0 +1,107 @@
+.. _howto-migrate-to-poetry:
+
+Migrate from the ``charm`` plugin to Poetry
```

Comment:
For consistency with existing docs and docs best practices, we should use the form `Charm plugin` and `Python plugin`.